test_e2e.sh

```diff
@@ -248,6 +248,52 @@ TAG="$TESTCASE"--TAG
 # First sync
 echo "$TESTCASE 1" > "$REPO"/file
 git -C "$REPO" commit -qam "$TESTCASE 1"
+git -C "$REPO" tag -f "$TAG" -m "$TESTCASE 1" >/dev/null
```

Comment:
Using -m in the tag command implies -a, so this test doesn't actually test non-annotated tags, from the git-tag man page.

>>If -m <msg> or -F <file> is given and -a, -s, and -u <keyid> are absent, -a is implied.